utils/v2/xyk.ts

```diff
@@ -0,0 +1,69 @@
+import { BN_ZERO } from "@mangata-finance/sdk";
+import { BN } from "@polkadot/util";
+import { api, Extrinsic } from "./setup";
+
+export class Xyk {
+  static createPool(
+    firstAsset: BN,
+    firstAssetAmount: BN,
+    secondAsset: BN,
+    secondAssetAmount: BN
+  ): Extrinsic {
+    return api.tx.xyk.createPool(
+      firstAsset,
+      firstAssetAmount,
+      secondAsset,
+      secondAssetAmount
+    );
+  }
+
+  static mintLiquidity(
+    firstAsset: BN,
+    secondAsset: BN,
+    firstAssetAmount: BN,
+    expectedSecondAssetAmount: BN = new BN(Number.MAX_SAFE_INTEGER)
+  ): Extrinsic {
+    return api.tx.xyk.mintLiquidity(
+      firstAsset,
+      secondAsset,
+      firstAssetAmount,
+      expectedSecondAssetAmount
+    );
+  }
+
+  static burnLiquidity(
+    firstAsset: BN,
+    secondAsset: BN,
+    assetAmount: BN
+  ): Extrinsic {
+    return api.tx.xyk.burnLiquidity(firstAsset, secondAsset, assetAmount);
```

Comment:
does this mean that we are not using SDK anymore?

Reply:
SDK uses Promises as results,
for batching we need plain Extrinsics
so yes, not using the SDK in most cases